### PR TITLE
feat: Add sidecars to deployment

### PR DESCRIPTION
**Pull request by nkreiger**
## Description

Adding ability to specify sidecars in the deployment, default is none.

## References

Used example:

https://github.com/bitnami/charts/tree/main/bitnami/keycloak 

For defining sidecars, because I noticed this chart already leverages a couple other bitnami packages.

## Review Checklist
- [ X ] I have clicked on ["allow edits by maintainers"](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork).
- [  ] I have added documentation for new/changed functionality in this PR or in a PR to [openfga.dev](https://github.com/openfga/openfga.dev) [Provide a link to any relevant PRs in the references section above]
- [ X  ] The correct base branch is being used, if not `main`
- [ ] I have added tests to validate that the change in functionality is working as expected

### CODE REVIEW

File: charts/openfga/values.yaml [hunk 1]
```diff
@@ -161,3 +161,15 @@ nodeSelector: {}
...
+# @param sidecars Add additional sidecar containers to the pods
+# Example:
+# sidecars:
+#   - name: your-image-name
+#     image: your-image
+#     imagePullPolicy: Always
+#     ports:
+#       - name: portname
+#         containerPort: 1234
+sidecars: []
```
Comment (jon-whit) [inline charts/openfga/values.yaml:174]: Please add to `values.schema.json` as well.

File: charts/openfga/values.yaml [hunk 2]
```diff
@@ -161,3 +161,15 @@ nodeSelector: {}
...
+migrateSidecars: []
```
Comment (jon-whit) [inline charts/openfga/values.yaml:175]: Let's nest this `migrate.sidecars`

e.g.

```yaml
migrate:
  sidecars: []
```

File: charts/openfga/values.yaml (hunk 1 above)
Comment (nkreiger) [reply charts/openfga/values.yaml:174]: added

File: charts/openfga/values.yaml (hunk 2 above)
Comment (nkreiger) [reply charts/openfga/values.yaml]: nested!

File: charts/openfga/templates/job.yaml [hunk 3]
```diff
@@ -31,6 +31,9 @@ spec:
                   name: "{{ .Values.datastore.uriSecret }}"
                   key: "uri"
             {{- end }}
+        {{- if .Values.migrateSidecars }}
```
Comment (jon-whit) [inline charts/openfga/templates/job.yaml:34]: This should be `migrate.sidecars`.

File: charts/openfga/templates/job.yaml [hunk 4]
```diff
@@ -31,6 +31,9 @@ spec:
...
+        {{- include "common.tplvalues.render" ( dict "value" .Values.sidecars "context" $) | nindent 8 }}
```
Comment (jon-whit) [inline charts/openfga/templates/job.yaml:35]: This should be `.Values.migrate.sidecars` not `.Values.sidecars`

**Review by jon-whit — CHANGES_REQUESTED** [review]
There are a few issues with the recent changes. Please fix and then I'll approve 👍 

Looks good otherwise.

**Comment by nkreiger** [comment]
> Needs a rebase on `main` and there are a few issues with the recent changes. Please fix and then I'll approve 👍
> 
> Looks good otherwise.

thanks @jon-whit apologies for the laziness on that update, just rebased and re-checked.

**Review by jon-whit — APPROVED** [review]

**Pull request merged**